operator/watchers/cilium_podippool.go

```diff
@@ -19,6 +28,107 @@ type PooledAllocatorProvider interface {
 	DeletePool(ctx context.Context, pool *cilium_v2alpha1.CiliumPodIPPool) error
 }
 
+const (
+	poolKeyIPv4CIDRs    = "ipv4-cidrs"
+	poolKeyIPv4MaskSize = "ipv4-mask-size"
+	poolKeyIPv6CIDRs    = "ipv6-cidrs"
+	poolKeyIPv6MaskSize = "ipv6-mask-size"
+)
+
+// parsePoolSpec parses a pool spec string in the form
+// "ipv4-cidrs:172.16.0.0/16,172.17.0.0/16;ipv4-mask-size:24".
+func parsePoolSpec(poolString string) (cilium_v2alpha1.IPPoolSpec, error) {
+	fields := strings.Split(strings.ReplaceAll(poolString, " ", ""), ";")
+
+	var ipv4CIDRs, ipv6CIDRs []cilium_v2alpha1.PoolCIDR
+	var ipv4MaskSize, ipv6MaskSize uint64
+
+	for _, field := range fields {
+		key, value, ok := strings.Cut(field, ":")
+		if !ok {
+			return cilium_v2alpha1.IPPoolSpec{}, fmt.Errorf("invalid number of key delimiters in pool spec %s", poolString)
+		}
+		switch key {
+		case poolKeyIPv4CIDRs:
+			for _, cidr := range strings.Split(value, ",") {
+				_, err := netip.ParsePrefix(cidr)
+				if err != nil {
+					return cilium_v2alpha1.IPPoolSpec{}, fmt.Errorf("invalid value for key %q: %w", poolKeyIPv4CIDRs, err)
+				}
+				ipv4CIDRs = append(ipv4CIDRs, cilium_v2alpha1.PoolCIDR(cidr))
+			}
+		case poolKeyIPv4MaskSize:
+			mask, err := strconv.ParseUint(value, 10, 64)
```

Comment:
not a real point, but its a bit odd we use uint64 for these field.

Reply:
Changed to a uint8.